src/main/java/org/iotdata/domain/function/tags/MapAbnormalHeartRateEvent.java

```diff
@@ -0,0 +1,39 @@
+package org.iotdata.domain.function.tags;
+
+import static org.apache.jena.sparql.expr.NodeValue.makeDateTime;
+import org.apache.jena.sparql.expr.NodeValue;
+
+/**
+ * Function used to detect abnormal heart rate events (< 60 and > 100)
+ */
+public class MapAbnormalHeartRateEvent extends MapAbstractEvent {
+	static final Integer LOW = 60;
+	static final Integer HIGH = 100;
+
+	public MapAbnormalHeartRateEvent(final Object ...params) {
+		super(params);
+	}
+
+	@Override
+	public String getName() {
+		return "mapHighHeartRateEventTime";
+	}
+
+	@Override
+	public NodeValue exec(final NodeValue identifier, final NodeValue timeStamp, final NodeValue heartRate) {
```

Comment:
Look at the comments within MapAbstractEvent - after applying them, you would be able to get rid of this exec :)